langstream-agents/langstream-agent-s3/src/main/java/ai/langstream/agents/s3/S3Source.java

```diff
@@ -49,32 +51,29 @@
 
 @Slf4j
 public class S3Source extends AbstractAgentCode implements AgentSource {
+
+    protected static ObjectMapper configMapper =
+            new ObjectMapper().configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+
     private String bucketName;
     private MinioClient minioClient;
     private final Set<String> objectsToCommit = ConcurrentHashMap.newKeySet();
     private int idleTime;
 
     public static final String ALL_FILES = "*";
-    public static final String DEFAULT_EXTENSIONS_FILTER = "pdf,docx,html,htm,md,txt";
     private Set<String> extensions = Set.of();
 
     @Override
     public void init(Map<String, Object> configuration) throws Exception {
-        bucketName = configuration.getOrDefault("bucketName", "langstream-source").toString();
-        String endpoint =
-                configuration
-                        .getOrDefault("endpoint", "http://minio-endpoint.-not-set:9090")
-                        .toString();
-        String username = configuration.getOrDefault("access-key", "minioadmin").toString();
```

Comment:
this is an old style agent, now all the new code uses ConfigurationUtils, how can we deal with it ? move everything to ObjectMapper ?